### PR TITLE
Fix incorrect field type issue

### PR DESCRIPTION
**Pull request by metrue**
- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] The PR targets the `latest` branch.
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [x] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

**Comment by metrue** [comment]
https://github.com/Azure/azure-sdk-for-go/issues/5339

File: services/recoveryservices/mgmt/2017-07-01/backup/models.go [hunk 1]
```diff
@@ -566,15 +567,15 @@ func PossibleJobStatusValues() []JobStatus {
...
 const (
 	// JobSupportedActionCancellable ...
-	JobSupportedActionCancellable JobSupportedAction = "Cancellable"
+	JobSupportedActionCancellable JobSupportedAction = iota
 	// JobSupportedActionInvalid ...
-	JobSupportedActionInvalid JobSupportedAction = "Invalid"
+	JobSupportedActionInvalid
 	// JobSupportedActionRetriable ...
-	JobSupportedActionRetriable JobSupportedAction = "Retriable"
+	JobSupportedActionRetriable
```
Comment (metrue) [inline services/recoveryservices/mgmt/2017-07-01/backup/models.go]: Changes need to confirm from Azure team,  the response from Azure API is like this, 

```
{
  "value": [
    {
      "id": "/Subscriptions/fc51e62f-aef2-49c3-b154-5b313db3f379/resourceGroups/Minghe_test/providers/Microsoft.RecoveryServices/vaults/backupvaultMinghetestdemo/backupJobs/c5529980-4735-4a7b-b360-5073cdef7185",
      "name": "c5529980-4735-4a7b-b360-5073cdef7185",
      "type": "Microsoft.RecoveryServices/vaults/backupJobs",
      "properties": {
        "jobType": "AzureIaaSVMJob",
        "duration": "01:09:58.9495507",
        "actionsInfo": [
          1
        ],
        "virtualMachineVersion": "Compute",
        "entityFriendlyName": "demo",
        "backupManagementType": "AzureIaasVM",
        "operation": "Backup",
        "status": "Completed",
        "startTime": "2019-09-19T03:07:26.0228043Z",
        "endTime": "2019-09-19T04:17:24.972355Z",
        "activityId": "535f4731-0a1b-4dd6-8139-01f66d258ce5"
      }
    }
  ]
}
```

The `actionsInfo` field of a job is a list of number, but there is no document about what a specific number means.

**Comment by jhendrixMSFT** [comment]
Note that this code is generated from [swagger files](https://github.com/azure/azure-rest-api-specs) so modifying it directly won't fix the underlying issue and will be lost in future updates.  @ArcturusZhang can you please follow up on this?

**Comment by ArcturusZhang** [comment]
Because this code is auto-generated from swagger, this fix will be overridden in next code generation. 
And in the [swagger](https://github.com/Azure/azure-rest-api-specs/blob/c383a94ea03a8364693023df2003f53c6127b7c8/specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json#L1107), this enum is clearly declared as string (actually all enum in azure is modeled as string). 
The root cause of this error lies in service side.
I submit a new [issue](https://github.com/Azure/azure-rest-api-specs/issues/7268) in the swagger repo about this.
Closing this pr.

**Comment by metrue** [comment]
So sad, almost a month passed,  no one is working on that issue https://github.com/Azure/azure-rest-api-specs/issues/7268

**Comment by ArcturusZhang** [comment]
> So sad, almost a month passed, no one is working on that issue [Azure/azure-rest-api-specs#7268](https://github.com/Azure/azure-rest-api-specs/issues/7268)

Hi @metrue sorry about the service team does not responds with this issue. Have you tried other api-versions? There are 2016-12-01 and 2016-06-01 api version for `backup`.
And there is a workaround, I am working on this. After I complete, I will post the workaround here.

**Comment by metrue** [comment]
@ArcturusZhang 

There is not `JobsClient` in API version 2016-12-01 and  2016-06-01 as I remember.

**Comment by ArcturusZhang** [comment]
> @ArcturusZhang
> 
> There is not `JobsClient` in API version 2016-12-01 and 2016-06-01 as I remember.

I see. The direct workaround is not very easy. Basically rewriting every method related with this `List` method. For your personal usage, you can always fork the whole SDK and apply your changes in your fork then use you fork as dependency.

**Comment by metrue** [comment]
That's what I am doing right now, but I think it's a must-fix bug for this SDK that should be fixed officially.

**Pull request closed without merging**